gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineConfig.java

```diff
@@ -102,41 +107,83 @@ public void execute(Task task) {
                 }
             });
 
+            Path checkstylePath = configDir.resolve(checkstylePathString);
+
             // Disable some checkstyle rules that clash with PJF
             if (BaselineFormat.palantirJavaFormatterState(rootProject) != FormatterState.OFF
                     || project.getPluginManager().hasPlugin("com.palantir.java-format-provider")) {
-                Path checkstyleXml = configDir.resolve("checkstyle/checkstyle.xml");
 
                 try {
-                    DocumentBuilderFactory builderFactory = DocumentBuilderFactory.newInstance();
-                    DocumentBuilder builder = builderFactory.newDocumentBuilder();
-
-                    InputSource inputSource = new InputSource(new FileReader(checkstyleXml.toFile()));
-                    inputSource.setEncoding("UTF-8");
-                    Document document = builder.parse(inputSource);
-
-                    XPathFactory xPathFactory = XPathFactory.newInstance();
-                    XPath xPath = xPathFactory.newXPath();
+                    Document document = getDocument(checkstylePath);
+                    XPath xPath = XPathFactory.newInstance().newXPath();
 
                     removeNode(document, xPath, "//module[@name='Indentation']");
                     removeNode(document, xPath, "//module[@name='ParenPad']");
                     removeNode(document, xPath, "//module[@name='LeftCurly']");
                     removeNode(document, xPath, "//module[@name='WhitespaceAround']");
 
-                    TransformerFactory transformerFactory = TransformerFactory.newInstance();
-                    Transformer transformer = transformerFactory.newTransformer();
-                    transformer.setOutputProperty(OutputKeys.ENCODING, "UTF-8");
-                    transformer.setOutputProperty(
-                            OutputKeys.DOCTYPE_PUBLIC, document.getDoctype().getPublicId());
-                    transformer.setOutputProperty(
-                            OutputKeys.DOCTYPE_SYSTEM, document.getDoctype().getSystemId());
-
-                    DOMSource source = new DOMSource(document);
-                    StreamResult result = new StreamResult(new FileWriter(checkstyleXml.toFile()));
-                    transformer.transform(source, result);
+                    writeEditedCheckstyle(document, checkstylePath);
+                } catch (Exception e) {
+                    throw new RuntimeException("Unable to patch " + checkstylePath, e);
+                }
+            }
+
+            if (!project.hasProperty(inclusiveCodeCheckOn)) {
```

Comment:
Thoughts on using a simple gradle plugin instead of a gradle.properties line? That way enablement works via `apply plugin: 'com.palantir.baseline-inclusive-language'`, where it's easier for us to leverage existing machinery (and state lives in the root build.gradle rather than being split between build.gradle and gradle.properties).

The palantir-java-format enablement has some legacy gradle properties, but in practice it's enabled by using `apply plugin: 'com.palantir.java-format'` at this point

Reply:
That sounds good! A plugin felt like a heavier lift than just a property, but if that's the standard then I will do that!

Reply:
To clarify, since this bit probably isn't obvious, the logic can be the same (since it's the update-config task that we're mutating in practice), but we'll use `hasPlugin(BaselineInclusiveLanguage.class)` instead of `hasProperty`.